lib/led/rgb.js

```diff
@@ -1,12 +1,117 @@
-var Led = require("./led"),
-  priv = new Map();
+var Board = require("../board.js");
+var nanosleep = require("../sleep.js").nano;
+
+var priv = new Map();
+
+var Controllers = {
+  DEFAULT: {
+    initialize: {
+      value: function(opts) {
+        RGB.colors.forEach(function(color, index) {
+          var pin = opts.pins[index];
+
+          if (!this.board.pins.isPwm(pin)) {
+            Board.Pins.Error({
+              pin: pin,
+              type: "PWM",
+              via: "Led.RGB"
+            });
+          }
+
+          this.io.pinMode(pin, this.io.MODES.PWM);
+          this.pins[index] = pin;
+        }, this);
+      }
+    },
+    write: {
+      value: function(colors) {
+        var state = priv.get(this);
+
+        RGB.colors.forEach(function(color, index) {
+          var pin = this.pins[index];
+          var value = colors[color];
+
+          if (state.isAnode) {
+            value = 255 - Board.constrain(value, 0, 255);
+          }
+
+          this.io.analogWrite(pin, value);
+        }, this);
+
+        Object.assign(state, colors);
+      }
+    }
+  },
+  PCA9685: {
+    COMMANDS: {
+      value: {
+        PCA9685_MODE1: 0x0,
+        PCA9685_PRESCALE: 0xFE,
+        LED0_ON_L: 0x6
+      }
+    },
+    initialize: {
+      value: function(opts) {
+        this.address = opts.address || 0x40;
+
+        if (!this.board.Drivers[this.address]) {
+          this.io.i2cConfig();
+          this.board.Drivers[this.address] = {
+            initialized: false
+          };
+
+          // Reset
+          this.io.i2cWrite(this.address, [this.COMMANDS.PCA9685_MODE1, 0x0]);
+          // Sleep
+          this.io.i2cWrite(this.address, [this.COMMANDS.PCA9685_MODE1, 0x10]);
+          // Set prescalar
+          this.io.i2cWrite(this.address, [this.COMMANDS.PCA9685_PRESCALE, 0x70]);
+          // Wake up
+          this.io.i2cWrite(this.address, [this.COMMANDS.PCA9685_MODE1, 0x0]);
+          // Wait 5 nanoseconds for restart
+          nanosleep(5);
+          // Auto-increment
+          this.io.i2cWrite(this.address, [this.COMMANDS.PCA9685_MODE1, 0xa1]);
+
+          this.board.Drivers[this.address].initialized = true;
+        }
+
+        RGB.colors.forEach(function(color, index) {
+          var pin = opts.pins[index];
+          this.pins[index] = pin;
+        }, this);
+      }
```

Comment:
It could. I don't think it buys us much, especially if controller initialization gets abstracted away in the future.